notify/notify.go

```diff
@@ -242,21 +297,21 @@ func (n *DedupingNotifier) hasUpdate(alert *types.Alert, last *types.Notificatio
 	return true
 }
 
-// Notify implements the Notifier interface.
-func (n *DedupingNotifier) Notify(ctx context.Context, alerts ...*types.Alert) error {
+// ExtractNewNotifies filters out the notifications that shall be sent
```

Comment:
I'm not completely happy with this design especially since we use returning `nil` as a special case to not continue the execution of the notification...